dvc/project.py

```diff
@@ -388,7 +389,9 @@ def run(self,
         all_stages = self.stages()
 
         self._check_cwd_specified_as_output(cwd, all_stages)
-        self._check_output_duplication(stage.outs, all_stages)
+
+        # Generate graph including the new stage to check for errors
+        self.graph(stages=all_stages + [stage])
```

Comment:
You used to have `if not overwrite` check here in the previous patch version :) We should either remove stage file when it is allowed to be overwritten or pop old stage from all_stages here first and then build a graph on all_stages + current stage.

Otherwise, you will get OutputDuplicationError whenever you try to overwrite the existing stage file with the same outputs. Let's add a test for this too.

Reply:
:+1: Added a related comment down below.